### ISSUE TITLE
Curried function with Word.andb gives incorrect result

### ISSUE THREAD

**Issue by stutonk** [polyml/polyml#133]
Best way to illustrate this is with the following REPL transcript.
```
Poly/ML 5.8 Release
> Word.andb(0w0, 0w1);
val it = 0wx0: word
> Word.andb(0w1, 0w0);
val it = 0wx0: word
> fun f (a, b) = Word.andb(a, b);
val f = fn: word * word -> word
> f (0w0, 0w1);
val it = 0wx0: word
> f (0w1, 0w0);
val it = 0wx0: word
> fun g a b = Word.andb(a, b);
val g = fn: word -> word -> word
> g 0w0 0w1;
val it = 0wx1: word
> g 0w1 0w0;
val it = 0wx0: word
```

**Comment by dcjm** [comment]
This has been fixed in GIT master 889f7c37 .  Note that this currently requires "`make compiler`".

**Comment by stutonk** [comment]
Confirmed. Thank you!
